During installation, if a swap partition exists, enable swap.

phoepsilonix opened the pull request:
If there is no swap at all when the partition of the installation destination device is mounted, if a Swap partition exists, enable swap. target_env_call cannot be used because it is before the files are copied to the installation destination. host_env_process_output cannot be used because the return value cannot be obtained. To avoid reinventing the wheel as much as possible, use subprocess.check_output.

dalto8 commented on src/modules/mount/main.py:
```diff
@@ -149,6 +149,21 @@ def get_btrfs_subvolumes(partitions):
+def enable_swap_partition():
+    swap_is_exist=subprocess.check_output(["swapon","-s"], universal_newlines=True)
+    if ( swap_is_exist == "" ):
+        all_partitions=subprocess.check_output(["lsblk", "-l", "-f", "-n", "-p"], universal_newlines=True).split("\n")
```
The list of partitions is already available in `partitions` if you read above a little in the code.  That should be used instead of a call to `lsblk`.

Using this same data structure, you should also ensure the partition is "claimed".  Otherwise we could overwrite data in a swap partition the user had not selected to use.  There are examples of using this in the fstab module if you want a reference.

dalto8 commented on src/modules/mount/main.py:
```diff
@@ -149,6 +149,21 @@ def get_btrfs_subvolumes(partitions):
+                swap = q[0]
+                try:
+                    subprocess.check_call(["swapon", swap])
+                except Exception as e:
```
This is too broad.  You should apply proper exception handling for the explicit exceptions that could be thrown.

dalto8 commented on src/modules/mount/main.py:
```diff
@@ -149,6 +149,21 @@ def get_btrfs_subvolumes(partitions):
+                    subprocess.check_call(["swapon", swap])
+                except Exception as e:
+                    try:
+                        subprocess.check_call(["mkswap", swap])
```
We shouldn't blindly call `mkswap` here.  `mkswap` should have already been run by the partition module.

dalto8 commented on src/modules/mount/main.py:
```diff
@@ -149,6 +149,21 @@ def get_btrfs_subvolumes(partitions):
     return btrfs_subvolumes
 
+def enable_swap_partition():
+    swap_is_exist=subprocess.check_output(["swapon","-s"], universal_newlines=True)
```
Don't use `subprocess`, instead use the built-in Calamares functions as described in the modules readme.

dalto8 reviewed (changes requested):
This looks much better than the previous PR.  Just a few things we need to touch up here.

phoepsilonix committed:
During installation, if a swap partition exists, enable swap.
If there is no swap at all when the partition of the installation destination device is mounted,
if a Swap partition exists, enable swap.
